cmake/modules/AddSwift.cmake

```diff
@@ -612,61 +629,69 @@ function(_add_swift_library_single target name)
+          ARCHIVE_OUTPUT_DIRECTORY_${config_upper} ${config_lib_dir}/${SWIFTLIB_SINGLE_SUBDIR})
+      endforeach()
+    endif()
 
-  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
-    set(install_name_dir "@rpath")
+    if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+      set(install_name_dir "@rpath")
 
-    if(SWIFTLIB_SINGLE_IS_STDLIB)
-      # Always use @rpath for XCTest.
-      if(NOT "${module_name}" STREQUAL "XCTest")
-        set(install_name_dir "${SWIFT_DARWIN_STDLIB_INSTALL_NAME_DIR}")
+      if(SWIFTLIB_SINGLE_IS_STDLIB)
+        # Always use @rpath for XCTest.
+        if(NOT "${module_name}" STREQUAL "XCTest")
+          set(install_name_dir "${SWIFT_DARWIN_STDLIB_INSTALL_NAME_DIR}")
+        endif()
       endif()
-    endif()
-
-    set_target_properties("${target}"
-      PROPERTIES
-      INSTALL_NAME_DIR "${install_name_dir}")
-  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" AND NOT "${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
-    set_target_properties("${target}"
-      PROPERTIES
-      INSTALL_RPATH "$ORIGIN:/usr/lib/swift/linux")
-  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Cygwin")
-    set_target_properties("${target}"
-      PROPERTIES
-      INSTALL_RPATH "$ORIGIN:/usr/lib/swift/windows")
-  endif()
 
-  set_target_properties("${target}" PROPERTIES BUILD_WITH_INSTALL_RPATH YES)
-  set_target_properties("${target}" PROPERTIES FOLDER "Swift libraries")
+      set_target_properties("${target}"
+        PROPERTIES
+        INSTALL_NAME_DIR "${install_name_dir}")
+    elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" AND NOT "${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
+      set_target_properties("${target}"
+        PROPERTIES
+        INSTALL_RPATH "$ORIGIN:/usr/lib/swift/linux")
+    elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Cygwin")
+      set_target_properties("${target}"
+        PROPERTIES
+        INSTALL_RPATH "$ORIGIN:/usr/lib/swift/windows")
+    endif()
 
-  # Configure the static library target.
-  # Set compile and link flags for the non-static target.
-  # Do these LAST.
+    set_target_properties("${target}" PROPERTIES BUILD_WITH_INSTALL_RPATH YES)
+    set_target_properties("${target}" PROPERTIES FOLDER "Swift libraries")
+  endif() # if(add_dynamic_library_target)
 
-  set(target_static)
+  # Next, we conditionally add a static library target.
   if(SWIFTLIB_SINGLE_IS_STDLIB AND SWIFT_BUILD_STATIC_STDLIB)
-    set(target_static "${target}-static")
+    set(add_static_library_target TRUE)
+  elseif(SWIFTLIB_SINGLE_IS_SDK_OVERLAY AND SWIFT_BUILD_STATIC_SDK_OVERLAY)
+    set(add_static_library_target TRUE)
+  else()
+    set(add_static_library_target FALSE)
+  endif()
+  if(add_static_library_target)
+    # Callers of `add_swift_library()` rely on the fact that it adds a target
+    # named `${target}`. Normally the dynamic library is given that name.
+    # However, if we are *only* building the static library, then it is the
+    # one given the name `${target}`.
```

Comment:
Same issue here -- the unsuffixed library can be static if it is a part of the compiler.